synapse/lib/cell.py

```diff
@@ -1851,18 +1852,31 @@ async def isCellActive(self):
         return self.isactive
 
     async def setCellActive(self, active):
+
+        if active == self.isactive:
+            return
+
         self.isactive = active
 
         if self.isactive:
+            self.activebase = await s_base.Base.anit()
             self._fireActiveCoros()
             await self._execCellUpdates()
             await self.initServiceActive()
         else:
             await self._killActiveCoros()
+            await self.activebase.fini()
+            self.activebase = None
             await self.initServicePassive()
 
         await self._setAhaActive()
 
+    def runActiveTask(self, coro):
+        # an API for active coroutines to use when running an
+        # ephemeral task which should be automatically torn down
+        # if the cell becomes inactive
+        return self.activebase.schedCoro(coro)
```

Comment:
We could also use the following istead of adding a new method:

`self.addActiveCoro(coro, base=self.activebase)`

That would allow us to treat the active tasks as part of our activecoros and let them be torn down and removed from the list of activecoros upon being set to inactive.

Reply:
The `runActiveTask` method is different from the `addActiveCoro` routine as it's implemented for one-shot functions rather than functions which expect to be restarted while the cell is active.